File: src/php/StandardAgent.php
<?php

declare(strict_types=1);

namespace AgentHarness;

class StandardAgent extends BaseAgent
{
    use HasHooks;
    use HasMiddleware;
    use UsesTools;
    use EmitsEvents;
    use HasShell;
    use HasSkills;

    public static function build(string $model): AgentBuilder
    {
        return new AgentBuilder($model);
    }
}

File: src/python/standard_agent.py
from src.python.base_agent import BaseAgent
from src.python.has_hooks import HasHooks
from src.python.has_middleware import HasMiddleware
from src.python.uses_tools import UsesTools
from src.python.emits_events import EmitsEvents
from src.python.has_shell import HasShell
from src.python.has_skills import HasSkills


class StandardAgent(BaseAgent, HasMiddleware, HasHooks, UsesTools, EmitsEvents, HasShell, HasSkills):
    pass
